### PR TITLE
Correct the exported cmake configuration file name

### PR DESCRIPTION
**Pull request by JackBoosY**
This PR will:
Fix the exported cmake file name to `wil-config.cmake` according to the [official documentation](https://cmake.org/cmake/help/latest/command/find_package.html#id6):
The command searches for a file called `<PackageName>Config.cmake` or `<lowercasePackageName>-config.cmake` for each name specified. 
And we should use `find_package(WIL CONFIG)` to find this library.

Related: https://github.com/microsoft/vcpkg/issues/21141.

### CODE REVIEW

**Comment by dg0yt** [comment]
@JackBoosY It seems you changed the EOL, making it hard to spot the actual changes on GH.

**Comment by dg0yt** [comment]
With `wil-config.cmake`, it remains unclear what is the desired canonical usage: either 
~~~
find_package(WIL)
~~~
or
~~~
find_package(wil)
~~~
Only one form can be correct, setting the desired `<Pkg>_FOUND`.

The alternative name, e.g. `WILConfig.cmake` or `wilConfig.cmake` (no dash!) leaves no doubt.

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,12 +45,13 @@ target_include_directories(${PROJECT_NAME} INTERFACE
...
 
 # Install Package Configuration
-install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}_targets)
-install(EXPORT ${PROJECT_NAME}_targets
+string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
+install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME_LOWER}_targets)
+install(EXPORT ${PROJECT_NAME_LOWER}_targets
     NAMESPACE ${PROJECT_NAME}::
-    FILE ${PROJECT_NAME}-config.cmake
+    FILE ${PROJECT_NAME_LOWER}-config.cmake
```
Comment (JackBoosY) [inline CMakeLists.txt:52]:
```suggestion
    FILE ${PROJECT_NAME_LOWER}-config.cmake
```suggestion
    FILE ${PROJECT_NAME_LOWER}Config.cmake

```
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -45,12 +45,16 @@ target_include_directories(${PROJECT_NAME} INTERFACE
...
+```suggestion
+    FILE ${PROJECT_NAME_LOWER}Config.cmake
+
```
Comment (JackBoosY) [inline CMakeLists.txt:55]:
```suggestion
    FILE ${PROJECT_NAME_LOWER}Config.cmake
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -45,12 +45,13 @@ target_include_directories(${PROJECT_NAME} INTERFACE
...
 
 # Install Package Configuration
-install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}_targets)
-install(EXPORT ${PROJECT_NAME}_targets
+string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
+install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME_LOWER}_targets)
+install(EXPORT ${PROJECT_NAME_LOWER}_targets
     NAMESPACE ${PROJECT_NAME}::
-    FILE ${PROJECT_NAME}-config.cmake
+    FILE ${PROJECT_NAME_LOWER}Config.cmake
```
Comment (jonwis) [inline CMakeLists.txt:52]: Per your description, should this be `${PROJECT_NAME_LOWER}-config.cmake` OR `${PROJECT_NAME}Config.cmake`? This looks like a mix of the two that probably works, but isn't either of what you found.

Comment (dg0yt) [reply CMakeLists.txt:52]: What is your desired usage:
`find_package(WIL)`
or 
`find_package(wil)`
or even
`find_package(Wil)`
?

Comment (JackBoosY) [reply CMakeLists.txt:52]: Ping @jonwis for response.

Comment (JackBoosY) [reply CMakeLists.txt:52]: Ping again.

**Review by ChrisGuzak — APPROVED** [review]

**Review by ChrisGuzak — APPROVED** [review]

**Pull request merged**